### ISSUE TITLE
Unable to open on virtual machine

### ISSUE THREAD

**Issue by KaceCottam** [rizinorg/cutter#2894]
**Environment information**
* Operating System: Kali Linux Virtual Machine
* Cutter version: 2.0.5
* Obtained from:
  - [ ] Built from source
  - [X] Downloaded release from Cutter website or GitHub 
  - [ ] Distribution repository
* File format: AppImage

**Describe the bug**

It fails open the AppImage with the error:

```
The X11 connection broke: No error (code 0)
XIO:  fatal IO error 2 (No such file or directory) on X server "localhost:10.0"
      after 422 requests (422 known processed) with 0 events remaining.
```

**To Reproduce**

I am working in a Kali Linux Virtual Machine (running on VirtualBox) through an SSH Connection with X11 Forwarding turned on. I am able to verify that X11 Forwarding works by using the program: `xeyes`.

1. In an ssh terminal connected to Kali Linux,
2. Get the AppImage with the command `wget https://github.com/rizinorg/cutter/releases/download/v2.0.5/Cutter-v2.0.5-x64.Linux.AppImage`
3. Make it executable with the command `chmod +x ./Cutter-v2.0.5-x64.Linux.AppImage`
4. Run application with `./Cutter-v2.0.5-x64.Linux.AppImage`
5. See error

**Expected behavior**

The program is supposed to open.

**Screenshots**

![image](https://user-images.githubusercontent.com/28381193/153123603-5211731e-d92f-4715-9d4d-d97a221385cd.png)

**Additional context**

I am working in a Kali Linux Virtual Machine (running on VirtualBox) through an SSH Connection with X11 Forwarding turned on. I am able to verify that X11 Forwarding works by using the program: `xeyes`.

Information on the kali virtual machine:

- Output of `lsb_release -a`:
 ```
 No LSB modules are available.
Distributor ID:	Kali
Description:	Kali GNU/Linux Rolling
Release:	2021.4
Codename:	kali-rolling
```

Information on the host machine:

- Output of `lsb_release -a`:
```
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 21.04
Release:	21.04
Codename:	hirsute
```

note that this ubuntu release is actually kubuntu (kde desktop environment)

Running with the environment variable `QT_DEBUG_PLUGINS=1`:
```console
$ QT_DEBUG_PLUGINS=1 ./Cutter-v2.0.5-x64.Linux.AppImage
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforms" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforms/libqxcb.so"
Found metadata in lib /tmp/.mount_Cutterf5HRLH/usr/plugins/platforms/libqxcb.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.QPlatformIntegrationFactoryInterface.5.3",
    "MetaData": {
        "Keys": [
            "xcb"
        ]
    },
    "archreq": 0,
    "className": "QXcbIntegrationPlugin",
    "debug": false,
    "version": 331520
}


Got keys from plugin meta data ("xcb")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/bin/platforms" ...
loaded library "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforms/libqxcb.so"
loaded library "Xcursor"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/plugins/platformthemes" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/bin/platformthemes" ...
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so"
Found metadata in lib /tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QPlatformInputContextFactoryInterface.5.1",
    "MetaData": {
        "Keys": [
            "compose",
            "xim"
        ]
    },
    "archreq": 0,
    "className": "QComposePlatformInputContextPlugin",
    "debug": false,
    "version": 331520
}


Got keys from plugin meta data ("compose", "xim")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so"
Found metadata in lib /tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so, metadata=
{
    "IID": "org.qt-project.Qt.QPlatformInputContextFactoryInterface.5.1",
    "MetaData": {
        "Keys": [
            "ibus"
        ]
    },
    "archreq": 0,
    "className": "QIbusPlatformInputContextPlugin",
    "debug": false,
    "version": 331520
}


Got keys from plugin meta data ("ibus")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/bin/platforminputcontexts" ...
loaded library "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so"
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations" ...
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations/libqxcb-egl-integration.so"
Found metadata in lib /tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations/libqxcb-egl-integration.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.Xcb.QXcbGlIntegrationFactoryInterface.5.5",
    "MetaData": {
        "Keys": [
            "xcb_egl"
        ]
    },
    "archreq": 0,
    "className": "QXcbEglIntegrationPlugin",
    "debug": false,
    "version": 331520
}


Got keys from plugin meta data ("xcb_egl")
QFactoryLoader::QFactoryLoader() looking at "/tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations/libqxcb-glx-integration.so"
Found metadata in lib /tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations/libqxcb-glx-integration.so, metadata=
{
    "IID": "org.qt-project.Qt.QPA.Xcb.QXcbGlIntegrationFactoryInterface.5.5",
    "MetaData": {
        "Keys": [
            "xcb_glx"
        ]
    },
    "archreq": 0,
    "className": "QXcbGlxIntegrationPlugin",
    "debug": false,
    "version": 331520
}


Got keys from plugin meta data ("xcb_glx")
QFactoryLoader::QFactoryLoader() checking directory path "/tmp/.mount_Cutterf5HRLH/usr/bin/xcbglintegrations" ...
loaded library "/tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations/libqxcb-glx-integration.so"
The X11 connection broke: No error (code 0)
XIO:  fatal IO error 2 (No such file or directory) on X server "localhost:10.0"
      after 422 requests (422 known processed) with 0 events remaining.
QLibraryPrivate::unload succeeded on "/tmp/.mount_Cutterf5HRLH/usr/plugins/xcbglintegrations/libqxcb-glx-integration.so"
QLibraryPrivate::unload succeeded on "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforminputcontexts/libcomposeplatforminputcontextplugin.so"
QLibraryPrivate::unload succeeded on "/tmp/.mount_Cutterf5HRLH/usr/plugins/platforms/libqxcb.so"
QLibraryPrivate::unload succeeded on "Xcursor" (faked)
```

**Comment by ret2libc** [comment]
Is either a .appimage issue or a non-cutter issue at all I guess. Would you be able to try to compile cutter directly and see if the x forwarding works that way (without the appimage, I mean)?

**Comment by KaceCottam** [comment]
It looks like compiling directly with the default options on release works as expected! I noticed that my version of kali does not have the `qt5-default` package (it seems that it has been depreciated in `hirsuite`), so I had to get all of the libraries manually.
I should note though, that even after installing all of the dependencies for building, the AppImage still does not seem to work-- so it must not be that I was missing some kind of dependency or something. I will continue using the compiled version, I just thought I would let y'all know!
